custom_components/tuya_local/devices/madimack_eco_heatpump_9kW.yaml

```diff
@@ -0,0 +1,130 @@
+name: Madimack Eco Heatpump 9kW
+products:
+  - id: 3z98f9dmfwtieuc7
+    manufacturer: Madimack
+    model: Eco 9kW
+
+entities:
+  - entity: climate
+    dps:
+      - id: 1
+        type: boolean
+        name: hvac_mode
+        mapping:
+          - dps_val: false
+            value: "off"
+          - dps_val: true
+            value: "heat"
+      - id: 2
+        type: string
+        name: mode
+        mapping:
+          - dps_val: heating
+            value: heat
+          - dps_val: cold
+            value: cool
+          - dps_val: auto
+            value: heat_cool
+      - id: 4
+        type: integer
+        name: temperature
+        range:
+          min: 18
+          max: 40
+      - id: 6
+        type: string
+        name: temperature_unit
+        mapping:
+          - dps_val: c
+            value: C
+          - dps_val: f
+            value: F
+      - id: 102
+        type: integer
+        name: current_temperature
+      - id: 5
+        type: string
+        name: preset_mode
+        mapping:
+          - dps_val: boost
+            value: boost
+          - dps_val: silence
+            value: sleep
+          - dps_val: power
+            value: boost
+
+  # Fault code as a SENSOR (no binary_sensor) – name must be 'sensor'
```

Comment:
This comment is incorrect. The convention for this integration is to turn fault sensors into a binary sensor to indicate whether there is a fault or not, with the fault code as an attribute.

A sensor with the raw fault code is not very useful, as the user still needs to decode the fault code to get any meaning from it.

Most devices have an example of this.